sdk/smx.ts

```diff
@@ -36,6 +40,47 @@ class SMXEvents {
       .withStateMachine({ currentPacket: new Uint8Array() }, handlePacket);
 
     // this.otherReports$.onValue((value) => console.log("Packet: ", value));
+
+    this.otherReports$
+    .filter((e) => e.type === 'host_cmd_finished')
+    .onValue((e) => console.log("Cmd Finished"));
+
+    const finishedCommand$ = this.otherReports$
+      .filter((e) => e.type === 'host_cmd_finished')
+      .map((e) => e.type === 'host_cmd_finished');
+
+    this.startedSend$ = new Bacon.Bus<boolean>();
+
+    // true means "it's ok to send", false means "don't send"
+    const okSend$ = new Bacon.Bus<boolean>()
+      .merge(finishedCommand$)  // Returns true when host_cmd_finished
+      .merge(this.startedSend$.not())  // Return false when starting to send
+      .toProperty(true);
+
+    // Main USB Output
+    this.output$ = new Bacon.Bus<Array<number>>();
+
+    // Config writes should only happen at most once per second. 
+    const configOutput$ = this.output$
+      .filter((e) => e[0] === API_COMMAND.WRITE_CONFIG_V5)
+      .throttle(1000);
+
+    const otherOutput$ = this.output$
+      .filter((e) => e[0] !== API_COMMAND.WRITE_CONFIG_V5);
+
+    const combinedOutput$ = new Bacon.Bus<Array<number>>()
+      .merge(configOutput$)
+      .merge(otherOutput$)
+      .bufferingThrottle(100)
+      .takeWhile(okSend$)
+      .doAction(_ => this.startedSend$.push(true))
+      .onValue(async (value) => await this.writeToHID(value));
```

Comment:
I can't get this to work without a bufferingThrottle.

I think what happens is `okSend$` is true, so it lets one thing through, it then pushes `startedSend$` so `okSend$` becomes false, but by the time  `okSend$` becomes true again from a packet finishing, `combinedOutput$` has 2 events lined up in the queue, so it then tries to send them both out at the same time. I tried `bufferingCount(1)` but that didn't help. Not really too sure how to get this to work how we want it quite yet. A little closer though.

Reply:
yeah, I think we do need some kind of buffer on it. may need to play with bacon a bit more to better understand the best pattern to use here

Reply:
Based on[ the docs for `takeWhile`](https://baconjs.github.io/api3/classes/eventstream.html#takewhile):
> Takes while given predicate function holds true, and then **ends**.

The ends part is significant, because when a stream ends, I don't think it ever emits events again. So, either the docs are wrong on this, or something else mysterious is happening. Also, when looking at what seems to be the inverse of `takeWhile`, [the `holdWhen` operator](https://baconjs.github.io/api3/classes/eventstream.html#holdwhen):
> Pauses and buffers the event stream if last event in valve is truthy. **All buffered events are released when valve becomes falsy.**

This "all events are released" behavior seems like the undesirable behavior you were observing that we need to work around with `bufferingThrottle`?